addons/GME/Scripts/Game/GME/Helpers/GME_VehicleHelper.c

```diff
@@ -0,0 +1,272 @@
+		if (!catalogue)
+			return null;
+		
+		if (uiInfo.HasEntityLabel(EEditableEntityLabel.VEHICLE_HELICOPTER))
+			catalogue.GetEntityListWithLabel(EEditableEntityLabel.TRAIT_HELI_CREW, entries);
+		
+		if (entries.IsEmpty())
+			catalogue.GetEntityListWithLabel(EEditableEntityLabel.ROLE_RIFLEMAN, entries);
+		
+		if (entries.IsEmpty())
+			return null;
+	
+		Resource res = Resource.Load(entries[0].GetPrefab());
+		if (!res.IsValid())
+			return null;
+	
+		EntitySpawnParams params = new EntitySpawnParams();
+			params.TransformMode = ETransformMode.WORLD;
+			params.Transform[3] = vehicle.GetOrigin();
+	
+		SCR_BaseCompartmentManagerComponent compartmentManager = SCR_BaseCompartmentManagerComponent.Cast(vehicle.FindComponent(SCR_BaseCompartmentManagerComponent));
+		if (!compartmentManager)
+			return null;
+	
+		array<BaseCompartmentSlot> slots = {};
+		compartmentManager.GetCompartmentsOfTypes(slots, compartmentTypes);
+	
+		SCR_AIGroup group = GME_GroupHelper.CreateEmptyGroup(params.Transform[3], faction);
+	
+		foreach (BaseCompartmentSlot slot : slots)
+		{
+			SCR_ChimeraCharacter char = SCR_ChimeraCharacter.Cast(GetGame().SpawnEntityPrefab(res, vehicle.GetWorld(), params));
+			if (!char)
+				continue;
+			
+			AIControlComponent aiControlComponent = AIControlComponent.Cast(char.FindComponent(AIControlComponent));
+			if (!aiControlComponent)
+				continue;
+			
+			aiControlComponent.ActivateAI();
+			group.AddAgent(aiControlComponent.GetAIAgent());
+			
+			SCR_CompartmentAccessComponent compartmentAccess = SCR_CompartmentAccessComponent.Cast(char.GetCompartmentAccessComponent());
+			if (compartmentAccess)
+				compartmentAccess.MoveInVehicle(vehicle, slot.GetType(), true, slot);
+		}
+	
+		return group;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	static array<SCR_AIGroup> SpawnPassengers(Vehicle vehicle, array<ResourceName> passengerGroupPrefabNames, array<ECompartmentType> compartmentTypes = GME_VehicleHelper.DEFAULT_PASSENGER_COMPARTMENT_TYPES)
+	{
+		return GME_VehicleHelper_SpawnPassengersJob(vehicle, passengerGroupPrefabNames, compartmentTypes).GetGroups();
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	static int GetPassengerCount(Vehicle vehicle, array<ECompartmentType> compartmentTypes = GME_VehicleHelper.DEFAULT_PASSENGER_COMPARTMENT_TYPES)
+	{
+		int count = 0;
+		
+		SCR_BaseCompartmentManagerComponent compartmentManager = SCR_BaseCompartmentManagerComponent.Cast(vehicle.FindComponent(SCR_BaseCompartmentManagerComponent));
+		if (!compartmentManager)
+			return 0;
+		
+		array<BaseCompartmentSlot> compartmentSlots = {};
+		compartmentManager.GetCompartmentsOfTypes(compartmentSlots, compartmentTypes);
+		
+		foreach (BaseCompartmentSlot slot : compartmentSlots)
+		{
+			if (slot.IsOccupied())
+				count++;
+		}
```

Comment:
I assume that if you just count the units in the vehicle without checking each seat, you will have no way of figuring out if passenger or crew?

Reply:
The `GetOccupants` method also iterates over all compartment slots and as you said, you can't filter for passengers.

Reply:
Fair. Thought as much. Thanks for the clear up :)